### PR TITLE
Handle enhanced #line directive in DirectiveTriviaSyntax.DirectiveNameToken

### PR DESCRIPTION
**Pull request by cston**
... and add several missing tests.

Test plan: https://github.com/dotnet/roslyn/issues/54509

### CODE REVIEW

**Comment by RikkiGibson** [comment]
There's no imperative to get this into preview2, right?

File: src/Compilers/CSharp/Portable/Syntax/DirectiveTriviaSyntax.cs [hunk 1]
```diff
@@ -44,6 +44,8 @@ public SyntaxToken DirectiveNameToken
                         return ((UndefDirectiveTriviaSyntax)this).UndefKeyword;
                     case SyntaxKind.LineDirectiveTrivia:
                         return ((LineDirectiveTriviaSyntax)this).LineKeyword;
+                    case SyntaxKind.LineSpanDirectiveTrivia:
+                        return ((LineSpanDirectiveTriviaSyntax)this).LineKeyword;
```
Comment (jcouv) [inline src/Compilers/CSharp/Portable/Syntax/DirectiveTriviaSyntax.cs:48]: Consider adding a direct compiler API test, since public API

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Test/Syntax/Diagnostics/LineSpanDirectiveTests.cs [hunk 2]
```diff
@@ -545,15 +550,18 @@ static void Main()
         B();
 #line (1, 1) - (1, 100) ""b.txt""
         C();
+#line (2, 3) - (2, 4) 9 ""a.txt""
+        D();
     }
     static void A() { }
     static void B() { }
     static void C() { }
+    static void D() { }
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Syntax/Diagnostics/LineSpanDirectiveTests.cs:559]: Does this cause the test to exhibit the bug fixed in this PR?

**Review by RikkiGibson — APPROVED** [review]

File: src/Compilers/CSharp/Test/Syntax/Diagnostics/LineSpanDirectiveTests.cs (hunk 2 above)
Comment (cston) [reply src/Compilers/CSharp/Test/Syntax/Diagnostics/LineSpanDirectiveTests.cs:559]: No, this test change is unrelated to the bug fix, and is only here to test a more interesting case than the line above.

**Pull request merged**